src/nih_wayfinding/app/scripts/routing/directions-service.js

```diff
@@ -72,6 +101,110 @@
                     return 'glyphicon-remove-circle';
             }
         }
+
+        function defaultDate() {
+            var now = new Date();
+            var day = now.getDay();
+            day = day < 10 ? '0' + day : day;
+            var month = now.getMonth() + 1;
+            month = month < 10 ? '0' + month : month;
+            var year = now.getYear() + 1900;
+            return month + '-' + day + '-' + year;
+        }
+
+        /**
+         * Transform OTP response to our GeoJson stub
+         * @param  {Object} otpResponse
+         * @return {geojson}    NIH geojson expected response
+         */
+        function transformOtpToGeoJson(otpResponse) {
+
+            var itineraries = otpResponse.plan.itineraries;
+            var itinerary = itineraries[0];
+            var leg = itinerary.legs[0];
+
+            // Foreach leg in legs
+            // get steps as points
+            var steps = _.map(leg.steps, function (step) {
+                return stepToPoint(step);
+            });
+            var numSteps = steps.length;
+
+            // Get legGeometry as feature collection of points
+            var legPoints = L.PolylineUtil.decode(leg.legGeometry.points);
+            var stepCollection = turf.featurecollection(_.map(legPoints, function (point) {
+                return turf.point([point[1], point[0]]);
+            }));
+
+            // Loop each of the legGeometry points, and add them to the previous step until we hit
+            //      a legGeometry point that is nearest to the next step. At that point, save off
+            //      the existing line points to the first step LineString and attach properties to it.
```

Comment:
Oh yuck.  Is the response coming back with just one enormous linestring?  There should be a better way.  I think there may be a query option like `showIntermediateSteps` or something that might give back individual geometries.  If not, perhaps it would be worth it to modify the response to break up the geometries.